service/history/replication/task_executor.go

```diff
@@ -174,7 +170,7 @@ func (e *taskExecutorImpl) handleActivityTask(
 	ctx, cancel := e.newTaskContext(ctx, attr.NamespaceId)
 	defer cancel()
 
-	err = e.historyEngine.SyncActivity(ctx, request)
+	_, err = e.shardContext.GetHistoryClient().SyncActivity(ctx, request)
```

Comment:
Will there be additional cost when we always go through history client? Should we still go through history engine if the request is for the local shard?

Reply:
we can certainly add additional wrapper, but the ROI is low?

Reply:
actually, if we decide to do perf optimization, the optimization should be done within the wrapper of client

Reply:
> actually, if we decide to do perf optimization, the optimization should be done within the wrapper of client

Yes. There will be additional network cost but I don't think it is a concern at this point. I can also add an additional check for now and move it to the wrapper of client.

Reply:
Ok Let's leave it for now and address it in case it becomes an issue in prod.